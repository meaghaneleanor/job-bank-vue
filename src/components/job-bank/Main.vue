<template>
  <div>
    <h1>Ontario job bank</h1>
    <JobResult />
  </div>
</template>

<script>
import JobResult from './JobResult.vue';

export default {
  components: {
    JobResult
  }    
};
</script>

