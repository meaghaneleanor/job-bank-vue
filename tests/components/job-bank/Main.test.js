import { shallowMount } from '@vue/test-utils';
import $i18n from '../../test.utils.js';

import Main from '../../../src/components/job-bank/Main.vue';

const defaultOptions = {
  mocks: {
    $i18n,
    jobsObject: {}
  },
  stubs: ['JobResult'],
};

const createComponent = (options = defaultOptions) => shallowMount(Main, options);

describe('Main.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = createComponent();
  });

  describe('snapshot', () => {
    it('should display the expected html', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('computed', () => {
    describe('.jobsObject', () => {
      let options = defaultOptions;
      options.methods = {
        getJobs: jest.fn().mockImplementation((lang) => lang)
      };

      describe('When the locale is changed', () => {
        it('should update the computed return', () => {
          $i18n.locale = 'blah';
          expect(options.methods.getJobs).toBeCalledWith($i18n.locale);
          expect(wrapper.vm.jobsObject).toEqual('blah');
        });
      });
    });
  });
});
