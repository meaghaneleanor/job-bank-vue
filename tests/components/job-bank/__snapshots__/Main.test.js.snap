// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main.vue snapshot should display the expected html 1`] = `"<div><h1>Ontario job bank</h1> </div>"`;
